decimal.go

```diff
@@ -773,6 +774,24 @@ func (d *Decimal) Scan(src interface{}) error {
 	}
 }
 
+var jsonNull = []byte("null")
+
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+// Adheres to the convention where []byte("null") is a no-op.
+func (d *Decimal) UnmarshalJSON(b []byte) error {
```

Comment:
Let's use the same method as the big.Int type: https://github.com/golang/go/blob/9d17e175e07a32f919e122faf617e09f9c9726da/src/math/big/intmarsh.go#L62

This also means we can remove the JSON methods completely since they can use the Text methods just fine. We also need both marshal and unmarshal.